util/templates.py

```diff
@@ -33,6 +37,12 @@ def render_configured_template(request, course, exercise, post_url, default=None
         template = default
     else:
         raise ConfigError("Missing \"template\" in exercise configuration.")
+    if "personalized" in exercise and exercise["personalized"]:
```

Comment:
I think this preparation should be moved into _exercise_context as well (request can be added as parameter). Even better the _exercise_context could invoke a "context function" in personalized.py that sets up the required directories and context.